addons/crm/models/crm_lead.py

```diff
@@ -1183,12 +1193,14 @@ def _creation_subtype(self):
 
     def _track_subtype(self, init_values):
         self.ensure_one()
-        if 'stage_id' in init_values and self.probability == 100 and self.stage_id:
+        if ('stage_id' in init_values or 'lost_reason' in init_values) and self.probability == 100 and self.stage_id:
             return self.env.ref('crm.mt_lead_won')
-        elif 'active' in init_values and self.probability == 0 and not self.active:
+        elif 'lost_reason' in init_values and not self.active:
             return self.env.ref('crm.mt_lead_lost')
         elif 'stage_id' in init_values:
             return self.env.ref('crm.mt_lead_stage')
+        elif 'team_id' not in init_values and self.active:
```

Comment:
We can use `if` instead of `elif` as re are returning inside (which is also more readable)